Fixes #31683 - content-export list can search by content view id

ofedoren opened the pull request:
@chris1984, this should fix the issue and potentially future ones.

Tests are passing locally, although I'm not sure if the logic is not broken for other commands that use/have `--content-view-version[-id]` option. Needs more manual testing with such options in other commands (I don't have needed environment though).

The fix assumes that both `--content-view[-id]` and `--content-view-version[-id]` can be used independently. Also I'd kindly ask reviewer to try this command with both options (ideally not using ids) and different combinations of those to be sure the command works as expected. I suggest to test on some data to see if the output corresponds to the data.

chris1984 reviewed (commented):
Resolves the main issue we were hitting:

```bash
[root@nightly ~]# hammer  content-export list --content-view-id 2
---|--------------------|----------------------------------------------------------------------------------|----------------------|-------------------------|-------------------------|------------------------
ID | DESTINATION SERVER | PATH                                                                             | CONTENT VIEW VERSION | CONTENT VIEW VERSION ID | CREATED AT              | UPDATED AT             
---|--------------------|----------------------------------------------------------------------------------|----------------------|-------------------------|-------------------------|------------------------
1  | partha             | /var/lib/pulp/exports/Default_Organization/view/2.0/partha/2021-01-26T15-44-4... | view 2.0             | 3                       | 2021-01-26 20:44:48 UTC | 2021-01-26 20:44:48 UTC
---|--------------------|----------------------------------------------------------------------------------|----------------------|-------------------------|-------------------------|------------------------
```

Still going through and doing some more testing on other areas.

chris1984 commented on lib/hammer_cli_katello/id_resolver.rb:
```diff
@@ -137,7 +139,7 @@ def content_view_version_id(options)
           .sort { |a, b| a['version'].to_f <=> b['version'].to_f }.last['id']
       else
-        pick_result(results, @api.resource(:content_view_versions))['id']
+        nil
```
Lets add this so Rubocop is happy:
`# rubocop:disable Style/EmptyElse`

ofedoren replied:
Thanks, fixed.

chris1984 reviewed (commented):
@ofedoren 

Tested it really well with a bunch of stuff and everything worked really good:

### List packages by Content View Version:
```bash
[vagrant@hammer hammer-cli-katello]$ hammer package list --content-view-version-id 3
---|-------------------------------|---------------------------
ID | FILENAME                      | SOURCE RPM                
---|-------------------------------|---------------------------
18 | bear-4.1-1.noarch.rpm         | bear-4.1-1.src.rpm        
6  | camel-0.1-1.noarch.rpm        | camel-0.1-1.src.rpm       
11 | cat-1.0-1.noarch.rpm          | cat-1.0-1.src.rpm         
7  | cheetah-1.25.3-5.noarch.rpm   | cheetah-1.25.3-5.src.rpm
---|-------------------------------|--------------------------
```
### List file by Content View Version:
```bash
[vagrant@hammer hammer-cli-katello]$ hammer file list --content-view-version="view 2.0"
---|------|-----
ID | NAME | PATH
---|------|-----
1  | hi   | hi  
---|------|-----
```
### List SRPM by Content View Version:
```bash
hammer srpm list --content-view-version-id 2
Warning: An error occured while loading module hammer_cli_foreman_remote_execution.
---|---------------------|---------------------------------------
ID | NAME                | FILENAME                              
---|---------------------|---------------------------------------
1  | geronimo-validation | geronimo-validation-1.1-11.el7.src.rpm
---|---------------------|---------------------------------------
```
Tried with `--content-view-version="view 2.0"` on both master and with your pr and I get the same result, so nothing broke:
`Error: At least one of options --content-view-id, --content-view is required.`

### Content View Version show using both --content-view-id and name:

```bash
[vagrant@hammer hammer-cli-katello]$ hammer content-view version show --content-view-id 2 --id 2
Id:                     2
Name:                   view 1.0
Version:                1.0
Description:            
Content View ID:        2
Content View Name:      view
Content View Label:     view
Lifecycle Environments: 

Repositories:           
 1) Id:    3
    Name:  Partha
    Label: Partha
```
```bash
[vagrant@hammer hammer-cli-katello]$ hammer content-view version show --content-view view --id 2
Id:                     2
Name:                   view 1.0
Version:                1.0
Description:            
Content View ID:        2
Content View Name:      view
Content View Label:     view
Lifecycle Environments: 

Repositories:           
 1) Id:    3
    Name:  Partha
    Label: Partha
```

### Activation Key create using both --content-view-id and name: 

```bash
[vagrant@hammer hammer-cli-katello]$ hammer activation-key create --name test1 --content-view-id 2 --organization-id 1 --lifecycle-environment-id 1
Activation key created.
```
```bash
[vagrant@hammer hammer-cli-katello]$ hammer activation-key create --name test2 --content-view view --organization-id 1 --lifecycle-environment-id 1
Warning: An error occured while loading module hammer_cli_foreman_remote_execution.
Activation key created.
```

### Deleted a Content View Version using both --content-view-id and name: 

```bash
[vagrant@hammer hammer-cli-katello]$ hammer content-view version delete --content-view-id 3 --id 5
[..........................................................................................................................................................................................................] [100%]
```
```bash
[vagrant@hammer hammer-cli-katello]$ hammer content-view version delete --content-view food --id 4
[..........................................................................................................................................................................................................] [100%]
```

The only thing I found that was different with the patch was the following:

`hammer content-view version show --content-view-id 2`

Before patch:

`Error: Found more than one content_view_version.`

With the patch:

```bash
[vagrant@hammer hammer-cli-katello]$ hammer content-view version show --content-view-id 2
Missing arguments for '--id'.
```

Not sure if you want to try and fix that in this PR or merge this and revisit later with your other changes?

ofedoren commented:
@chris1984, updated and added new test data based on katello master branch in a separate commit. Also, thanks for great testing! :heart:

Regarding `Missing arguments for '--id'.`: I don't think it's caused by this patch. After some runs I found (at least it seems so) that it fails with this error in case if content view version's id matches content view's id or if a content view has more than one version. Otherwise it seems to be working. I'd took a look at the logic in the API for those commands, but I really lack needed knowledge about content related stuff.

From hammer perspective it's totally okay to require `required` option/parameter mentioned in the API docs unless someone in the hammer code makes oposite decision.

chris1984 commented:
> @chris1984, updated and added new test data based on katello master branch in a separate commit. Also, thanks for great testing! ❤️
> 
> Regarding `Missing arguments for '--id'.`: I don't think it's caused by this patch. After some runs I found (at least it seems so) that it fails with this error in case if content view version's id matches content view's id or if a content view has more than one version. Otherwise it seems to be working. I'd took a look at the logic in the API for those commands, but I really lack needed knowledge about content related stuff.
> 
> From hammer perspective it's totally okay to require `required` option/parameter mentioned in the API docs unless someone in the hammer code makes oposite decision.

That reasoning works fine for me. I think this one is good to go. One last rubocop failure to fix. After that ACK:

````ruby
lib/hammer_cli_katello/id_resolver.rb:120:5: C: Use empty lines between method definitions.
    def content_view_version_id(options)
    ^^^

249 files inspected, 1 offense detected
```

ofedoren commented:
@chris1984, thanks, updated :)

chris1984 commented:
Merged sir :)
